lib/krikri/parsers/xml_parser.rb

```diff
@@ -61,9 +61,81 @@ def values?
 
       private
 
-      def get_child_nodes(name)
-        Krikri::Parser::ValueArray.new(@node.xpath("#{@node.path}/#{name}", @namespaces)
-          .map { |node| self.class.new(node, @namespaces) })
+      ##
+      # Return a Krikri::Parser::ValueArray of child nodes that match a
+      # name or "name expression" as interpreted by #xpath_predicate
+      #
+      # @see Krikri::JsonParser#get_child_nodes
+      # @see Krikri::XmlParser::Value#xpath_expression
+      # @see Krikri::XmlParser::Value#xpath_predicate
+      #
+      # @example
+      #    va = value.get_child_nodes('dc:subject')
+      #    va = value.get_child_nodes('dc:this|dc:that')
+      #
+      # @param name_exp [String]  Element name expression
+      # @return [Krikri::Parser::ValueArray]
+      def get_child_nodes(name_exp)
+        Krikri::Parser::ValueArray.new(
+          @node.xpath(xpath_expression(name_exp), @namespaces)
+            .map { |node| self.class.new(node, @namespaces) }
+        )
+      end
+
+      ##
+      # Return a string for an XPATH expression
+      #
+      # The resulting path string will produce the element that occurs first in
+      # the XML document, which matches one of the given names.
+      #
+      # @param name_exp [String]  Element name expression
+      # @return [String]
+      def xpath_expression(name_exp)
+        #
+        # First, we have to figure out the base path.
+        #
+        # An assumption is made that no parser is really going to be used in
+        # a mapping without a root path (as defined by the root_path
+        # parameter to its constructor).  The usual XPATH expression will
+        # tend to be the first one.  The predicate will not work with
+        # @node.path == '/*', which is the default when the root path is not
+        # specified.
+        #
+        # When the root path is specified, name values with the "xmlns" prefix
+        # will not work; since this is the default namespace, we want to refer
+        # to them as the un-prefixed name.  For example, inside an OAI-PMH
```

Comment:
Yeah, this definitely needs a spec, because `xmlns:` elements are currently resolving appropriately, e.g. in the [ESDN mapping](https://github.com/dpla/heidrun-mappings/blob/master/heidrun/esdn_mods.rb).